### PR TITLE
Implement quickfix to remove '.java' when KClass is expected

### PR DESCRIPTION
**Pull request by cypressious**
Fixes #KT-11078

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertClassToKClassFix.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        if (!super.isAvailable(project, editor, file)) return false
+
+        val expressionType = element.analyze(BodyResolveMode.PARTIAL).getType(element) ?: return false
+        if (!expressionType.isJClass()) return false
+
+        val children = element.children
+        if (children.size != 2) return false
+
+        val firstChild = children.first() as? KtExpression ?: return false
+        val firstChildType = firstChild.analyze(BodyResolveMode.PARTIAL).getType(firstChild) ?: return false
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertClassToKClassFix.kt]: You should reuse the BindingContext from the first analyze() call.

File: idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertClassToKClassFix.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        return firstChildType.isSubtypeOf(type)
+    }
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val lastChild = element.children.lastOrNull() ?: return
+        editor?.document?.replaceString(lastChild.startOffset - 1, lastChild.endOffset, "")
```
Comment (yole) [inline idea/src/org/jetbrains/kotlin/idea/quickfix/ConvertClassToKClassFix.kt]: The quickfix will do nothing if there is no editor. It's possible to perform the change through the PSI, and it's better to do it that way.

**Comment by cypressious** [comment]
@yole Done

**Comment by yole** [comment]
Merged manually. Thanks a lot!

**Pull request closed without merging**